src/Actions/CreateNewUser.php

```diff
@@ -81,9 +82,9 @@ private function getUserData(array $input): array
         return $userData;
     }
 
-    private function usernameRules(): array
+    private function usernameRules(): mixed
     {
-        $rules = ['required', 'string', 'max:255', 'unique:users', resolve(PoliteUsername::class)];
+        $rules = ['required', 'string', 'min:3', 'max:30', 'unique:users', resolve(PoliteUsername::class), resolve(AllowedCharactersUsername::class)];
```

Comment:
this will cause issues with email logins, as it will apply `AllowedCharactersUsername` on that too